effectful/ops/syntax.py

```diff
@@ -280,7 +280,7 @@ def infer_annotations(cls, sig: inspect.Signature) -> inspect.Signature:
         # pre-conditions
         assert cls._check_has_single_scope(sig)
         assert cls._check_no_typevar_overlap(sig)
-        assert cls._check_no_boundvars_in_result(sig)
+        # assert cls._check_no_boundvars_in_result(sig)
```

Comment:
Instead of disabling this check for all operations, can we reenable it and make the signatures of the offending operations compliant? This check corresponds to a case our alpha-renaming code doesn't yet cover, so we shouldn't stop enforcing it in general.

Reply:
Sorry, forgot about that. I think `named_distribution` doesn't pass this check, but I think its signature is correct.